### DISCUSSION TITLE
[data-importer] imported transactions with link to transaction id 0 and then broke the database

### DISCUSSION THREAD

**Discussion by marcsc13** [firefly-iii/firefly-iii#5721] [Firefly III] [answered]
Hello together,

thanks for all the hard work that you've put into this. It's awesome.

I made a stupid mistake when trying to import a csv with the latest data-importer.
I deleted transactions which where created in firefly because they had a broken link.
When hovering over a transaction in the transaction list, it showed 'transactions/show/0' and they were not clickable.
I assumed something went wrong during the import because the data-importer imported some of the transactions but then stopped with an error (I don't have the error message anymore, sry).
So I tried to delete them but they had no action button.
That was the time the stupidity started. 
I removed the transactions from the MariaDB `transactions` table beginning from the date of the first broken transaction.
There is no database dump, and no way to restore them as far as I know.

Now I am stuck with an error message of `FireflyIII\Support\Twig\AmountFormat::FireflyIII\Support\Twig\{closure}(): Argument #1 ($amount) must be of type string, null given, called in /var/www/html/storage/framework/views/twig/79/794cd6ee55fc68cdb1062accc77dfaa78c8ae5beb24efe33447451f676b0bc0b.php on line 601` when trying to access the transaction view. 
Some months work and the reconcile account view shows transactions till the date I deleted the broken transactions.

I tried to understand the database tables, but no foreign key constraints are violated 
It would be awesome if someone might be able to give me a hint. 
PHP is not in my repertoire and I was not sure if this would be a reason to open an issue. 
There is no problem with Firefly, I just broke it.

Thanks everyone in advance, Marc

**Comment by JC5** [comment, marked as answer]
Find the transactions in the table `transaction_journals` and delete them from there. That should fix it.

**Reply by marcsc13** [reply]
Thanks a lot, that fixed it
